pkg/encryption/cipher.go

```diff
@@ -91,50 +91,134 @@ func checkHmac(input, expected string) bool {
 	return false
 }
 
-// Cipher provides methods to encrypt and decrypt cookie values
-type Cipher struct {
-	cipher.Block
+// Cipher provides methods to encrypt and decrypt
+type Cipher interface {
+	Encrypt(value []byte) ([]byte, error)
+	Decrypt(ciphertext []byte) ([]byte, error)
 }
 
 // NewCipher returns a new aes Cipher for encrypting cookie values
-func NewCipher(secret []byte) (*Cipher, error) {
+// This defaults to the Base64 Cipher to align with legacy Encrypt/Decrypt functionality
+func NewCipher(secret []byte) (Cipher, error) {
+	cfb, err := NewCFBCipher(secret)
+	if err != nil {
+		return nil, err
+	}
+	return NewBase64Cipher(cfb)
+}
+
+type Base64Cipher struct {
+	Cipher Cipher
+}
+
+// NewBase64Cipher returns a new AES Cipher for encrypting cookie values
+// and wrapping them in Base64 -- Supports Legacy encryption scheme
+func NewBase64Cipher(c Cipher) (Cipher, error) {
+	return &Base64Cipher{Cipher: c}, nil
+}
+
+// Encrypt encrypts a value with AES CFB & base64 encodes it
+func (c *Base64Cipher) Encrypt(value []byte) ([]byte, error) {
+	encrypted, err := c.Cipher.Encrypt([]byte(value))
+	if err != nil {
+		return nil, err
+	}
+
+	return []byte(base64.StdEncoding.EncodeToString(encrypted)), nil
+}
+
+// Decrypt Base64 decodes a value & decrypts it with AES CFB
+func (c *Base64Cipher) Decrypt(ciphertext []byte) ([]byte, error) {
+	encrypted, err := base64.StdEncoding.DecodeString(string(ciphertext))
+	if err != nil {
+		return nil, fmt.Errorf("failed to decrypt cookie value %s", err)
+	}
+
+	return c.Cipher.Decrypt(encrypted)
+}
+
+type CFBCipher struct {
+	cipher.Block
+}
+
+// NewCFBCipher returns a new AES CFB Cipher
+func NewCFBCipher(secret []byte) (Cipher, error) {
 	c, err := aes.NewCipher(secret)
 	if err != nil {
 		return nil, err
 	}
-	return &Cipher{Block: c}, err
+	return &CFBCipher{Block: c}, err
 }
 
-// Encrypt a value for use in a cookie
-func (c *Cipher) Encrypt(value string) (string, error) {
+// Encrypt with AES CFB
+func (c *CFBCipher) Encrypt(value []byte) ([]byte, error) {
 	ciphertext := make([]byte, aes.BlockSize+len(value))
 	iv := ciphertext[:aes.BlockSize]
 	if _, err := io.ReadFull(rand.Reader, iv); err != nil {
-		return "", fmt.Errorf("failed to create initialization vector %s", err)
+		return nil, fmt.Errorf("failed to create initialization vector %s", err)
 	}
 
 	stream := cipher.NewCFBEncrypter(c.Block, iv)
-	stream.XORKeyStream(ciphertext[aes.BlockSize:], []byte(value))
-	return base64.StdEncoding.EncodeToString(ciphertext), nil
+	stream.XORKeyStream(ciphertext[aes.BlockSize:], value)
+	return ciphertext, nil
+}
+
+// Decrypt an AES CFB ciphertext
+func (c *CFBCipher) Decrypt(ciphertext []byte) ([]byte, error) {
+	if len(ciphertext) < aes.BlockSize {
+		return nil, fmt.Errorf("encrypted value should be "+
+			"at least %d bytes, but is only %d bytes",
+			aes.BlockSize, len(ciphertext))
```

Comment:
I'd prefer this to be one line, we don't need to strictly keep to 80 char line length (I realise this is how it was before but would you mind refactoring?)